src/presentation/user/user.dto.ts

```diff
@@ -0,0 +1,26 @@
+import { createZodDto } from 'nestjs-zod';
+import { z } from 'nestjs-zod/z';
+
+const AddUserSchema = z.object({
+  email: z.string().email(),
+  password: z
+    .string()
+    .min(8)
+    .max(255)
+    .regex(
+      /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[$*&@#])[0-9a-zA-Z$*&@#]{8,}$/,
+      {
+        message:
+          'The password must contain at least one uppercase letter, one lowercase letter, one number, and one special character',
```

Comment:
Isn't it necessary to inform that the pass needs at least 8 chars?

Reply:
No because there is a validation for that too, if the user try to put a password with non 8-digits it will throw a different error. The regex will apply just after the user insert 8 digits or more